### PR TITLE
Add chapter on password encryption

### PR DESCRIPTION
**Pull request by ianf77**
Controller password encryption only

Draft and publish docs based on KCS for database password encryption

Affects `titles/aap-operations-guide/`

https://issues.redhat.com/browse/AAP-10226

### CODE REVIEW

File: downstream/modules/platform/proc-encrypt-postgres-password.adoc [hunk 1]
```diff
@@ -0,0 +1,38 @@
+[id="proc-encrypt-postgres-password"]
+
+= Encrypting the Postgres password
+
+The following procedure replaces the plain text passwords with encrypted values. 
+Perform the following steps on each node in the cluster:
+
+. Procedure
```
Comment (ariordan-redhat) [inline downstream/modules/platform/proc-encrypt-postgres-password.adoc:8]:
```suggestion
.Procedure
```

File: downstream/modules/platform/proc-encrypt-postgres-password.adoc [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
++
+[NOTE]
+====
+The hash value in this step is the output value of `postgres_secret`. 
+====
+. The full `postgres.py` resembles the following:
++
+[literal, options="nowrap" subs="+quotes,attributes"]
+----
+# Ansible Tower database settings. from awx.main.utils import decrypt_value, get_encryption_key DATABASES = { 'default': { 'ATOMIC_REQUESTS': True, 'ENGINE': 'django.db.backends.postgresql', 'NAME': 'awx', 'USER': 'awx', 'PASSWORD': decrypt_value(get_encryption_key('value'),'$encrypted$AESCBC$Z0FBQUFBQmNONU9BbGQ1VjJyNDJRVTRKaFRIR09Ib2U5TGdaYVRfcXFXRjlmdmpZNjdoZVpEZ21QRWViMmNDOGJaM0dPeHN2b194NUxvQ1M5X3dSc1gxQ29TdDBKRkljWHc9PQ=='), 'HOST': '127.0.0.1', 'PORT': 5432, } }
```
Comment (msilmser) [inline downstream/modules/platform/proc-encrypt-postgres-password.adoc:37]: Replace `# Ansible Tower database settings` with `# Ansible Automation Platform controller database settings.

**Review by msilmser — APPROVED** [review]
/etc/tower/conf.d is still the correct directory, that was never renamed but I did add one comment to show how that file does display in recent versions below. approving this but that change should get made.

Thanks!

File: downstream/assemblies/platform/assembly-encrypting-plaintext-passwords.adoc [hunk 3]
```diff
@@ -0,0 +1,18 @@
+ifdef::context[:parent-context: {context}]
+
+[id="encrypting-plaintext-passwords"]
+= Encrypting plaintext passwords in {ControllerName} configuration files
+
+:context: encrypt-password
+
+Passwords stored in {ControllerName} configuration files are stored in plain text. 
+A user with access to the `/etc/tower/conf.d/` directory can view the passwords used to access the database.
+Accesxs to the directories is controlled with permissions, so they are protected, but some security findings deem this protection to be inadequate.
```
Comment (ariordan-redhat) [inline downstream/assemblies/platform/assembly-encrypting-plaintext-passwords.adoc:10]:
```suggestion
Access to the directories is controlled with permissions, so they are protected, but some security findings deem this protection to be inadequate.
```

File: downstream/modules/platform/proc-create-password-hashes.adoc [hunk 4]
```diff
@@ -0,0 +1,38 @@
+[id="proc-create-password-hashes"]
+
+= Creating PostgreSQL password hashes
+
+.Procedure
+. On your automation controller node, run the following:
++
+[literal, options="nowrap" subs="+quotes,attributes"]
+----
```
Comment (ariordan-redhat) [inline downstream/modules/platform/proc-create-password-hashes.adoc:9]: Should the code be preceded with a `$` prompt?

**Review by ariordan-redhat — CHANGES_REQUESTED** [review]
A typo, but the rest of the comments can be ignored

File: downstream/modules/platform/proc-create-password-hashes.adoc (hunk 4 above)
Comment (ianf77) [reply downstream/modules/platform/proc-create-password-hashes.adoc:9]: I don't know. It doesn't look much like command line code. At least not Linux. I think it's Python.

**Review by ariordan-redhat — APPROVED** [review]
LGTM

**Pull request merged**